core/common/buffer.hpp

```diff
@@ -208,6 +208,35 @@ namespace kagome::common {
     Buffer &putRange(const T &begin, const T &end);
   };
 
+  /**
+   * @brief override operator<< for all streams except std::ostream
+   * @tparam Stream stream type
+   * @param s stream reference
+   * @param buffer value to encode
+   * @return reference to stream
+   */
+  template <
+      class Stream,
+      typename = std::enable_if_t<!std::is_same<Stream, std::ostream>::value>>
```

Comment:
Wouldn't it be better to use std::decay here?

Reply:
In what way? I don't get it.

Reply:
For Stream in std::is_same

Reply:
It is not necessary here.